homework_1.js

```diff
@@ -0,0 +1,153 @@
+// 1
+// Функция принимает массив пользователей. Исходные данные см. ниже. Возвращает объект, состоящий из двух полей: female и male. Данные
+//   поля являются массивами, в которых содержаться пользователи, подходящие по категории gender соответственно.
+//   Также, вместо двух полей first_name и last_name у каждого из объектов должно быть поле fullName в котором будут объеденины убранные поля
+// (first_name и last_name). Количество пользователей может быть не ограничено.
+
+const users = [
+    {
+      id: 1,
+      first_name: 'Jeanette',
+      last_name: 'Penddreth',
+      email: 'jpenddreth0@census.gov',
+      gender: 'female',
+      ip_address: '26.58.193.2',
+    },
+    {
+      id: 2,
+      first_name: 'Petr',
+      last_name: 'Jackson',
+      email: 'gfrediani1@senate.gov',
+      gender: 'male',
+      ip_address: '229.179.4.212',
+    },
+  ]
+  
+  const func = (arr) => {
+    const arrWithFullName = arr.map((item) => {
+      item.fullName = `${item.first_name} ${item.last_name}`
+      delete item.last_name
+      delete item.first_name
+      return item
+    })
+  
+    const women = arrWithFullName.filter((item) => item.gender === 'female')
+    const men = arrWithFullName.filter((item) => item.gender === 'male')
+    return { women: women, men: men }
```

Comment:
1) if I ask you to add 10 more genders, let's imagine how your function will grow up.
2) Your function mutate entire data. That's not good. 

My suggestion will be - rewrite your function with reduce and think about scalability